src/helper_functions.cpp

```diff
@@ -32,7 +32,11 @@
 
 std::string string_from_unterminated(const char* data, size_t data_length)
 {
+#if defined(OS_SOLARIS)
+    const size_t StringLength = strlen(data);
```

Comment:
Please don't do this, the ẁhole point of `string_from_unterminated` is to give you the length of a string that *might* not have a proper termination via `\0`. By using `strlen()` you'll get a guaranteed out of bounds read in that case.